src/plugins/data/public/query/query_string/query_string_manager.ts

```diff
@@ -154,10 +154,40 @@ export class QueryStringManager {
    */
   public setQuery = (query: Partial<Query>) => {
     const curQuery = this.query$.getValue();
-    const newQuery = { ...curQuery, ...query };
+    let newQuery = { ...curQuery, ...query };
     if (!isEqual(curQuery, newQuery)) {
-      // Add to recent datasets if dataset has changed
+      // Check if dataset changed and if new dataset has language restrictions
       if (newQuery.dataset && !isEqual(curQuery.dataset, newQuery.dataset)) {
+        // Get supported languages for the dataset
+        const supportedLanguages = this.datasetService
+          .getType(newQuery.dataset.type)
+          ?.supportedLanguages(newQuery.dataset);
+
+        // If we have supported languages and current language isn't supported
+        if (supportedLanguages && !supportedLanguages.includes(newQuery.language)) {
+          // Get initial query with first supported language and new dataset
+          newQuery = this.getInitialQuery({
+            language: supportedLanguages[0],
+            dataset: newQuery.dataset,
+          });
+
+          // Show warning about language change
+          showWarning(this.notifications, {
+            title: i18n.translate('data.languageChangeTitle', {
+              defaultMessage: 'Language Changed',
```

Comment:
```suggestion
              defaultMessage: 'Language changed',
```

Reply:
i'm not sure if this will retrigger requiring approvals so will do this in fast follow! thanks